shared/domains/chelonia/chelonia.js

```diff
@@ -1642,50 +1646,135 @@ export default (sbp('sbp/selectors/register', {
+        })
+      } else {
+        // If no initial data provided, perform a GET `fetch` to get the current
+        // data and CID. Then, `onconflict` will be used to merge the current
+        // and new data.
+        response = await this.config.fetch(url, {
+          headers: new Headers([[
+            'authorization', buildShelterAuthorizationHeader.call(this, contractID)
+          ]]),
+          signal: this.abortController.signal
+        })
+      }
+      // The `resolveData` function is tasked with computing merged data, as in
+      // merging the existing stored values (after a conflict or initial fetch)
+      // and new data. The return value indicates whether there should be a new
+      // attempt at storing updated data (if `true`) or not (if `false`)
+      const resolveData = async () => {
+        let currentValue
+        // Rationale:
+        //  * response.ok could be the result of `GET` (no initial data)
+        //  * 409 indicates a conflict because the height used is too old
+        //  * 412 indicates a conflict (precondition failed) because the data
+        //    on the KV store have been updated / is not what we expected
+        // All of these situations should trigger parsing the respinse and
+        // conlict resolution
+        if (response.ok || response.status === 409 || response.status === 412) {
+          const serializedData = await response.json()
+          currentValue = parseEncryptedOrUnencryptedMessage.call(this, {
+            contractID,
+            serializedData,
+            meta: key
+          })
+        // Rationale: 404 and 410 both indicate that the store key doesn't exist.
+        // These are not treated as errors since we could still set the value.
+        } else if (response.status !== 404 && response.status !== 410) {
+          throw new ChelErrorUnexpectedHttpResponseCode('[kv/set] Invalid response code: ' + response.status)
+        }
+        const result = await (onconflict: Function)({
+          contractID,
+          key,
+          failedData: data,
+          status: response.status,
+          etag: response.headers.get('x-cid') || response.headers.get('etag'),
+          get currentData () {
+            return currentValue?.data
+          },
+          currentValue
+        })
+        if (!result) return false
+
+        data = result[0]
+        ifMatch = result[1]
+        return true
+      }
+      // Initial 'resolveData', used when the selector is called with no data,
+      // in which case a `GET` is performed before attempting to save
+      if (data === undefined) {
+        if (!hasOnconflict) {
+          throw TypeError('onconflict required with empty data')
+        }
+        // This is only for the initial case; the logic is replicated below
+        // for subsequent iterations that require conflic resolution.
+        if (await resolveData()) {
+          continue
+        } else {
+          break
+        }
+      }
```

Comment:
Could you please move the contents of this `if` block inside of the `else` branch on line `1691`?